### PR TITLE
[IMP] core: few http/readonly cleanups after merge

### PR DESCRIPTION
**Pull request by Julien00859**
The development of the readonly feature spanned the entirety of 2024. In the last few months / weeks of develoment, the teams were eager to speed up the review process to merge it for saas-17.1, it was decided to stop polishing the code and to instead focus on the last few blocking changes.

We are now a few months later, the tension has settle down and it is possible to take a bit of time to polish a few things. Like to add missing docstrings, to isolate a few bits inside dedicated methods and overall to try to improve readability. Every function (including the docstring) should now fit on one page (~40 lines).

One of the goal of 16.0 HTTPocalypse was to reduce the size of the callstack to get smaller tracebacks. Like we did during for that other PR, we replaced a few closures by `functools.partial` when possible. Using `functools.partial` instead of a closure / a lambda makes it invisible in the callstack and doesn't add complexity.

---

Run the tests with

```
--test-tags database_breaking:TestHttpRegistry
```

**/!\\** it will duplicate your current database

### CODE REVIEW

File: odoo/http.py [hunk 1]
```diff
@@ -1782,71 +1803,89 @@ def _serve_db(self):
...
+        """
+        self.params = self.get_http_params()
+        response = self.registry['ir.http']._serve_fallback()
+        if response:
+            self.registry['ir.http']._post_dispatch(response)
+            return response
 
-            return self._transactioning(_serve_ir_http, readonly=ro)
+        no_fallback = NotFound()
+        no_fallback.__context__ = not_found  # During handling of {not_found}, {no_fallback} occurred:
```
[the remark this answers is not in the document]
Comment (Julien00859) [reply odoo/http.py:1881]: Well there are two exceptions... and those two exceptions are NotFound... but different kinds of NotFound... Do you have a suggestion in mind?

**Review by rco-odoo — APPROVED** [review]
Good work!

**Comment by rco-odoo** [comment]
@robodoo override=ci/security

The spotted potential issues are justified in the context of low-level hacks for testing errors with HTTP routing.

**Comment by Julien00859** [comment]
@robodoo r+

**Comment by Julien00859** [comment]
@robodoo rebase-ff

**Pull request closed without merging**